test/sass/functions_test.rb

```diff
@@ -1368,12 +1378,21 @@ def test_map_get_checks_type
   end
 
   def test_map_merge
+    # These are the version where the second argument is a map
     assert_equal("(foo: 1, bar: 2, baz: 3)",
       perform("map-merge((foo: 1, bar: 2), (baz: 3))").to_sass)
     assert_equal("(foo: 1, bar: 2)",
       perform("map-merge((), (foo: 1, bar: 2))").to_sass)
     assert_equal("(foo: 1, bar: 2)",
       perform("map-merge((foo: 1, bar: 2), ())").to_sass)
+
+    # Here, we test when the second argument is an array of keys, plus a value
+    assert_equal("(a: b, c: d)",
+            perform("map-merge((a: b), (c: d))").to_sass)
+    assert_equal("(a: (b: c, d: e))",
+            perform("map-merge((a: (b: c)), a, (d: e))").to_sass)
+    assert_equal("(a: (d: e))",
+            perform("map-merge((a: (b: c)), (a: (d: e)))").to_sass)
```

Comment:
Test nesting deeper than actually exists, nesting through values that aren't maps, and nesting through empty maps. That goes for all of these functions that don't already test these cases.